packages/video_player/video_player_web/lib/video_player_web.dart

```diff
@@ -176,10 +176,38 @@ class _VideoPlayer {
         isInitialized = true;
         sendInitialized();
       }
+      if (isBuffering) {
+        isBuffering = false;
+        eventController.add(VideoEvent(eventType: VideoEventType.bufferingEnd));
+      }
+    });
+
+    videoElement.onCanPlayThrough.listen((dynamic _) {
+      if (isBuffering) {
+        isBuffering = false;
+        eventController.add(VideoEvent(eventType: VideoEventType.bufferingEnd));
+      }
+    });
+
+    videoElement.onPlaying.listen((dynamic _) {
+      if (isBuffering) {
+        isBuffering = false;
+        eventController.add(VideoEvent(eventType: VideoEventType.bufferingEnd));
+      }
+    });
+
+    videoElement.onWaiting.listen((dynamic _) {
+      if (!isBuffering) {
+        isBuffering = true;
+        eventController
+            .add(VideoEvent(eventType: VideoEventType.bufferingStart));
+        sendBufferingUpdate();
+      }
     });
 
     // The error event fires when some form of error occurs while attempting to load or perform the media.
     videoElement.onError.listen((Event _) {
+      isBuffering = false;
```

Comment:
Why do we not need to send the "bufferingEnd" event in this case?

Reply:
I checked, and it seems the android version doesn't send it either, it just fires the error event. So I guess this is good for feature parity.

Reply:
Error events on both these platforms will now also send the buffering end event also (only if was buffering previously) - made sense to send this as otherwise there could be a potential for an invalid buffering state when the player errors.